tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/Emitter.scala

```diff
@@ -345,11 +345,20 @@ final class Emitter private (semantics: Semantics, outputMode: OutputMode,
 
     // Class definition
     if (linkedClass.hasInstances && kind.isAnyNonNativeClass) {
+      val (linkedInlinedInit, linkedMemberMethods) =
+        classEmitter.extractInlinedInit(linkedClass)(classCache)
+
+      // JS constructor
+      val inlinedInit = linkedInlinedInit.map { m =>
+        val inlinedInitCache = classCache.getInlinedInitCache()
+        inlinedInitCache.getOrElseUpdate(m.version,
+            classEmitter.genInlinedInit(className, m.tree)(inlinedInitCache))
+      }
       val ctor = classTreeCache.constructor.getOrElseUpdate(
-          classEmitter.genConstructor(linkedClass)(classCache))
+          classEmitter.genConstructor(linkedClass, inlinedInit)(classCache))
```

Comment:
This is broken. If `inlinedInit` gets updated (i.e. the ctor body), but the class does not, this does not regenerate.